File: lib/models/app_settings.dart
/// Application settings
class AppSettings {
  final double sensitivity;
  final Duration idleTimeout;
  final bool autoReconnect;
  final int maxReconnectAttempts;

  AppSettings({
    this.sensitivity = 1.0,
    this.idleTimeout = const Duration(seconds: 60),
    this.autoReconnect = true,
    this.maxReconnectAttempts = 10,
  });

  Map<String, dynamic> toJson() {
    return {
      'sensitivity': sensitivity,
      'idleTimeout': idleTimeout.inSeconds,
      'autoReconnect': autoReconnect,
      'maxReconnectAttempts': maxReconnectAttempts,
    };
  }

  factory AppSettings.fromJson(Map<String, dynamic> json) {
    return AppSettings(
      sensitivity: (json['sensitivity'] as num?)?.toDouble() ?? 1.0,
      idleTimeout: Duration(seconds: json['idleTimeout'] as int? ?? 60),
      autoReconnect: json['autoReconnect'] as bool? ?? true,
      maxReconnectAttempts: json['maxReconnectAttempts'] as int? ?? 10,
    );
  }

  AppSettings copyWith({
    double? sensitivity,
    Duration? idleTimeout,
    bool? autoReconnect,
    int? maxReconnectAttempts,
  }) {
    return AppSettings(
      sensitivity: sensitivity ?? this.sensitivity,
      idleTimeout: idleTimeout ?? this.idleTimeout,
      autoReconnect: autoReconnect ?? this.autoReconnect,
      maxReconnectAttempts: maxReconnectAttempts ?? this.maxReconnectAttempts,
    );
  }

  @override
  String toString() {
    return 'AppSettings(sensitivity: $sensitivity, idleTimeout: $idleTimeout, autoReconnect: $autoReconnect, maxReconnectAttempts: $maxReconnectAttempts)';
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;

    return other is AppSettings &&
        other.sensitivity == sensitivity &&
        other.idleTimeout == idleTimeout &&
        other.autoReconnect == autoReconnect &&
        other.maxReconnectAttempts == maxReconnectAttempts;
  }

  @override
  int get hashCode {
    return sensitivity.hashCode ^
        idleTimeout.hashCode ^
        autoReconnect.hashCode ^
        maxReconnectAttempts.hashCode;
  }
}

File: macos/Runner/Models/AppSettings.swift
//
//  AppSettings.swift
//  RemoteTouch macOS
//
//  Application settings for macOS app
//

import Foundation

/// Application settings
struct AppSettings: Codable {
    let sensitivity: Double
    let idleTimeout: TimeInterval
    let autoReconnect: Bool
    let maxReconnectAttempts: Int
    
    init(
        sensitivity: Double = 1.0,
        idleTimeout: TimeInterval = 60.0,
        autoReconnect: Bool = true,
        maxReconnectAttempts: Int = 10
    ) {
        self.sensitivity = sensitivity
        self.idleTimeout = idleTimeout
        self.autoReconnect = autoReconnect
        self.maxReconnectAttempts = maxReconnectAttempts
    }
    
    enum CodingKeys: String, CodingKey {
        case sensitivity
        case idleTimeout
        case autoReconnect
        case maxReconnectAttempts
    }
    
    /// Encode to JSON data
    func toJSON() throws -> Data {
        let encoder = JSONEncoder()
        return try encoder.encode(self)
    }
    
    /// Decode from JSON data
    static func fromJSON(_ data: Data) throws -> AppSettings {
        let decoder = JSONDecoder()
        return try decoder.decode(AppSettings.self, from: data)
    }
    
    /// Create a copy with modified properties
    func copyWith(
        sensitivity: Double? = nil,
        idleTimeout: TimeInterval? = nil,
        autoReconnect: Bool? = nil,
        maxReconnectAttempts: Int? = nil
    ) -> AppSettings {
        return AppSettings(
            sensitivity: sensitivity ?? self.sensitivity,
            idleTimeout: idleTimeout ?? self.idleTimeout,
            autoReconnect: autoReconnect ?? self.autoReconnect,
            maxReconnectAttempts: maxReconnectAttempts ?? self.maxReconnectAttempts
        )
    }
}

// MARK: - CustomStringConvertible

extension AppSettings: CustomStringConvertible {
    var description: String {
        return "AppSettings(sensitivity: \(sensitivity), idleTimeout: \(idleTimeout), autoReconnect: \(autoReconnect), maxReconnectAttempts: \(maxReconnectAttempts))"
    }
}

// MARK: - Equatable

extension AppSettings: Equatable {
    static func == (lhs: AppSettings, rhs: AppSettings) -> Bool {
        return lhs.sensitivity == rhs.sensitivity &&
               lhs.idleTimeout == rhs.idleTimeout &&
               lhs.autoReconnect == rhs.autoReconnect &&
               lhs.maxReconnectAttempts == rhs.maxReconnectAttempts
    }
}

// MARK: - Hashable

extension AppSettings: Hashable {
    func hash(into hasher: inout Hasher) {
        hasher.combine(sensitivity)
        hasher.combine(idleTimeout)
        hasher.combine(autoReconnect)
        hasher.combine(maxReconnectAttempts)
    }
}
